src/main/java/core/basesyntax/dao/CarJdbcDaoImpl.java

```diff
@@ -0,0 +1,170 @@
+import core.basesyntax.model.Car;
+import core.basesyntax.model.Driver;
+import core.basesyntax.model.Manufacturer;
+import core.basesyntax.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class CarJdbcDaoImpl implements CarDao {
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT * FROM cars c "
+                + "JOIN manufacturers m on m.manufacturer_id = c.manufacturer_id "
+                + "JOIN cars_drivers cd on c.car_id = cd.car_id "
+                + "JOIN drivers d on d.driver_id = cd.driver_id "
+                + "WHERE d.driver_id = ? AND c.deleted = FALSE AND d.deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement carsByDriverStatement = connection.prepareStatement(query)) {
+            carsByDriverStatement.setLong(1, driverId);
+            ResultSet result = carsByDriverStatement.executeQuery();
+            List<Car> cars = new ArrayList<>();
+            while (result.next()) {
+                cars.add(getCar(result, connection));
+            }
+            return cars;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t get car by driver id " + driverId, e);
+        }
+    }
+
+    @Override
+    public Car create(Car car) {
+        String query = "INSERT INTO cars(model, manufacturer_id) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createdCar = connection
+                        .prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            createdCar.setString(1, car.getModel());
+            createdCar.setLong(2, car.getManufacturer().getId());
+            createdCar.executeUpdate();
+            ResultSet resultSet = createdCar.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t create car " + car, e);
+        }
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String query = "SELECT c.car_id, c.model, c.manufacturer_id, "
+                + "m.manufacturer_id, m.brand, m.country FROM cars c "
+                + "JOIN manufacturers m on m.manufacturer_id = c.manufacturer_id "
+                + "WHERE car_id = ? AND c.deleted = FALSE";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement gettingCarStatement = connection.prepareStatement(query)) {
+            gettingCarStatement.setLong(1, id);
+            ResultSet result = gettingCarStatement.executeQuery();
```

Comment:
```suggestion
            ResultSet resultSet = gettingCarStatement.executeQuery();
```
result is too abstract name

Reply:
Fixed